prow/plank/controller.go

```diff
@@ -524,42 +516,49 @@ func (c *Controller) syncTriggeredJob(pj prowapi.ProwJob, pm map[string]corev1.P
 	return c.prowJobClient.Patch(c.ctx, pj.DeepCopy(), ctrlruntimeclient.MergeFrom(&prevPJ))
 }
 
-// TODO: No need to return the pod name since we already have the
-// prowjob in the call site.
-func (c *Controller) startPod(pj prowapi.ProwJob) (string, string, error) {
+func (c *Controller) startPod(pj *prowapi.ProwJob) error {
 	buildID, err := c.getBuildID(pj.Spec.Job)
 	if err != nil {
-		return "", "", fmt.Errorf("error getting build ID: %v", err)
+		return fmt.Errorf("error getting build ID: %v", err)
 	}
 
-	pod, err := decorate.ProwJobToPod(pj, buildID)
+	// TODO: remove buildID from input args to ProwJobToPod
+	pj.Status.BuildID = buildID
+	pod, err := decorate.ProwJobToPod(*pj, buildID)
 	if err != nil {
-		return "", "", err
+		return err
 	}
 	pod.Namespace = c.config().PodNamespace
 
 	client, ok := c.buildClients[pj.ClusterAlias()]
 	if !ok {
-		return "", "", fmt.Errorf("unknown cluster alias %q", pj.ClusterAlias())
+		return fmt.Errorf("unknown cluster alias %q", pj.ClusterAlias())
 	}
 	err = client.Create(c.ctx, pod)
-	c.log.WithFields(pjutil.ProwJobFields(&pj)).Debug("Create Pod.")
+	c.log.WithFields(pjutil.ProwJobFields(pj)).Debug("Create Pod.")
 	if err != nil {
-		return "", "", err
+		return err
 	}
-	return buildID, pod.ObjectMeta.Name, nil
+	pj.Status.PodName = pod.ObjectMeta.Name
+	return nil
 }
 
 func (c *Controller) getBuildID(name string) (string, error) {
 	return pjutil.GetBuildID(name, c.totURL)
 }
 
 func getPodBuildID(pod *corev1.Pod) string {
+	if buildID, ok := pod.ObjectMeta.Labels[kube.ProwBuildIDLabel]; ok && buildID != "" {
```

Comment:
super nit: This is slightly more idiomatic:
```go
if buildID := pod.ObjectMeta.Labels[kube.ProwBuildIDLabel]; buildID != "" {
```

Reply:
Thank you for the review. Tide merged the PR before I could correct it.

Reply:
Yeah, I didn't want to block the PR on this since it is such a minor style change. It's just a small tip for the future.